bson/primitive/objectid.go

```diff
@@ -94,6 +94,18 @@ func IsValidObjectID(s string) bool {
 	return err == nil
 }
 
+// MarshalText returns the ObjectID as UTF-8-encoded text. Implementing this allows us to use ObjectID
```

Comment:
Consider adding a variable assignment that asserts that `primitive.ObjectID` implements `encoding.TextMarshaler` and `encoding.TextUnmarshaler`.

E.g.
```go
var _ encoding.TextMarshaler = ObjectID{}
var _ encoding.TextUnmarshaler = ObjectID{}
```